kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtPackageServiceImpl.java

```diff
@@ -126,136 +114,34 @@ public void uninstallDeploymentPackage(GwtXSRFToken xsrfToken, String packageNam
         }
     }
 
-    private Node getFirstNode(final Document doc, final String tagName) {
-        final NodeList elements = doc.getElementsByTagName(tagName);
-        if (elements.getLength() == 0) {
-            return null;
-        }
-        return elements.item(0);
-    }
-
-    private String getAttributeValue(NamedNodeMap attributes, String attribute) {
-        final Node node = attributes.getNamedItem(attribute);
-        if (node == null) {
-            return null;
-        }
-        return node.getNodeValue();
-    }
-
     @Override
     public GwtMarketplacePackageDescriptor getMarketplacePackageDescriptor(GwtXSRFToken xsrfToken, String nodeId)
             throws GwtKuraException {
         checkXSRFToken(xsrfToken);
 
-        GwtMarketplacePackageDescriptor descriptor = null;
-        URL mpUrl = null;
-        HttpsURLConnection connection = null;
-
+        GwtMarketplacePackageDescriptor descriptor = new GwtMarketplacePackageDescriptor();
         try {
-
-            mpUrl = new URL(String.format(MARKETPLACE_URL, nodeId));
-            connection = (HttpsURLConnection) mpUrl.openConnection();
-            connection.setSSLSocketFactory(this.sslManagerService.getSSLSocketFactory());
```

Comment:
Currently the console has a configuration property that allows to specify the `SslManagerService` that is used to fetch deployment packages from the marketlplace, this change would make it use the SslManagerService configured in the deployment agent instead.. We could add another `getMarketplacePackageDescriptor(String, SslManagerService)` to the DeploymentAgent that allows to specify the `SslManagerService` to restore the old behaviour

Reply:
Addressed in 08eca41ab8e2c7d638e0938176e0058bf65414e1